core/eth/contracts.go

```diff
@@ -1,23 +1,34 @@
 package eth
 
 import (
-	"chainlink/core/logger"
 	"fmt"
+	"math/big"
+	"reflect"
 	"strings"
 
+	"chainlink/core/logger"
+
 	"github.com/ethereum/go-ethereum/accounts/abi"
 	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
 	"github.com/gobuffalo/packr"
 	"github.com/pkg/errors"
 	"github.com/tidwall/gjson"
 )
 
+type Contract interface {
+	ABI() abi.ABI
+	GetMethodID(method string) ([]byte, error)
+	EncodeMessageCall(method string, args ...interface{}) ([]byte, error)
+	UnpackLog(out interface{}, event string, log Log) error
+}
+
 // Contract holds the solidity contract's parsed ABI
-type Contract struct {
-	ABI abi.ABI
+type contract struct {
+	abi abi.ABI
 }
```

Comment:
Maybe we should start avoiding this pattern?  Instead returning structs.  I've run into it a lot, and it seems like carry over from other languages, and its used a lot in this code base too.

Not sure how we could best deal with the name clash though.

Reply:
Lets discuss this at some point.  This pattern is already used in the code base, so it should not hold back these changes.

Reply:
What's wrong here @felder-cl ? Not sure I follow.